runtime-agent-full-run: wp-codebox monorepo npm install/build OOM-killed (status null) blocks all agent runs

## Summary

`runtime-agent-full-run.yml` agent runs fail deterministically at the runtime-dependency build: a `npm install` is killed by a signal (`status: null` → SIGKILL, i.e. OOM) on the GitHub-hosted runner, so the agent never executes and the run fails with `ERROR: results file missing or empty: .../runtime-agent-run/results.json`.

This was **latent since 2026-06-23** — the runner workflow itself couldn't parse (unquoted-colon YAML, fixed in #2001), so no run got past parse to expose this until now.

## Reproduction

Any consumer of `runtime-agent-full-run.yml` (e.g. `Automattic/build-with-wordpress` developer-docs). Deterministic — reproduced on two consecutive runs (28333429909, 28335270278), same failure point.

## Mechanism (root cause)

`setup-runtime.cjs` runs the wp-codebox runtime's materialization commands from `agent-runtimes/wp-codebox/wp-codebox.json`:

```json
"ci_materialization": {
  "checkout": { "repo": "Automattic/wp-codebox", "target": ".ci/wp-codebox" },
  "setup_commands": [ { "command": "npm", "args": ["install"], "cwd": ".ci/wp-codebox" } ],
  "build_commands": [ { "command": "npm", "args": ["run", "build"], "cwd": ".ci/wp-codebox" } ]
}
```

So it checks out the **full `Automattic/wp-codebox` monorepo** and runs `npm install` + `npm run build` on it. The `run()` helper (`runtime-agent-ci/lib/full-run-inputs.cjs:197`) uses `spawnSync(..., { stdio: 'inherit' })` and throws on `status !== 0`; `status: null` indicates the process was terminated by a signal (SIGKILL/OOM), not a non-zero exit. The wp-codebox monorepo install/build footprint appears to exceed the runner's memory.

## Log evidence

```
node .../setup-runtime.cjs
...composer installs (php deps)...
145 packages are looking for funding
npm install failed with status null      # <- killed by signal
##[error]Process completed with exit code 1.
ERROR: results file missing or empty: .../runtime-agent-run/results.json
```

## Fix direction (to investigate)

1. Pin down whether it is the `.ci/wp-codebox` `npm install`, the `npm run build`, or the `.ci/homeboy-extensions/wordpress` adapter install (`agent-runtimes/wp-codebox/lib/runtime-setup.cjs:11`) that is killed.
2. Reduce the build footprint and/or raise memory headroom, e.g.:
   - `npm install --no-audit --no-fund` and only the package(s) needed for the runtime bin, rather than the whole monorepo;
   - constrain parallelism / set `NODE_OPTIONS=--max-old-space-size=...` for `npm run build` (tsc -b across the monorepo is memory-heavy);
   - or consume a prebuilt/published wp-codebox CLI instead of building the monorepo from source in CI.

## Acceptance criteria

A `runtime-agent-full-run.yml` consumer run gets past `setup-runtime.cjs` (runtime deps install + build succeed) and the agent executes — no `status: null` kill, `results.json` is produced.

--- a/runtime-agent-ci/lib/full-run-inputs.cjs
+++ b/runtime-agent-ci/lib/full-run-inputs.cjs
@@ -195,8 +195,16 @@ function getByPath(value, expression) {
 
 function run(command, args, options = {}) {
   const result = spawnSync(command, args, { stdio: 'inherit', ...options });
+  // spawnSync returns status: null both when the process is killed by a signal
+  // and when it cannot be spawned at all (result.error, e.g. a missing cwd).
+  // Surface the real cause instead of a bare "status null".
+  if (result.error) {
+    const cwd = options.cwd ? ` (cwd: ${options.cwd})` : '';
+    throw new Error(`${command} ${args.join(' ')} failed to spawn${cwd}: ${result.error.message}`);
+  }
   if (result.status !== 0) {
-    throw new Error(`${command} ${args.join(' ')} failed with status ${result.status}`);
+    const detail = result.signal ? ` (signal ${result.signal})` : '';
+    throw new Error(`${command} ${args.join(' ')} failed with status ${result.status}${detail}`);
   }
 }
 

--- a/runtime-agent-ci/lib/runtime-provider-resolver.cjs
+++ b/runtime-agent-ci/lib/runtime-provider-resolver.cjs
@@ -248,12 +248,20 @@ function normalizeRuntimeId(runtimeId = DEFAULT_RUNTIME_ID, options = {}) {
 }
 
 function runtimeIdFromOptions(options = {}, env = process.env) {
+	// Mirror the env precedence used by setup-runtime.cjs
+	// (RUNTIME || RUNTIME_PROVIDER || BACKEND). Consumers commonly set only
+	// RUNTIME_PROVIDER; without honoring it here, materialize-dependencies
+	// resolves DEFAULT_RUNTIME_ID instead of the real runtime and never checks
+	// out the runtime provider repo, so its setup_commands later fail on a
+	// missing cwd.
 	return firstString(
 		options.runtimeId,
 		options.runtime_id,
 		options.runtime,
 		env.RUNTIME_ID,
 		env.RUNTIME,
+		env.RUNTIME_PROVIDER,
+		env.BACKEND,
 		DEFAULT_RUNTIME_ID
 	);
 }

